frontend/utils/distribution/group-income-distribution.js

```diff
@@ -1,29 +1,16 @@
 import { saferFloat } from '~/frontend/views/utils/currencies.js'
-import incomeDistribution from '~/frontend/utils/distribution/mincome-proportional.js'
-
-export default function groupIncomeDistribution ({ state, getters, monthstamp, adjusted }: Object): any {
-  // the monthstamp will always be for the current month. the alternative
+import parseMonthlyDistributionFromEvents from '~/frontend/utils/distribution/monthly-event-parser.js'
+import minimizeTotalPaymentsCount from '~/frontend/utils/distribution/payments-minimizer.js'
+export default function groupIncomeDistribution ({ state, getters, monthstamp, adjusted, minimizeTxns = false }: Object): any {
+  // the monthstamp will alwayparseProRatedDistributionFromEventss be for the current month. the alternative
   // is to allow the re-generation of the distribution for previous months,
   // but that approach requires also storing the historical mincomeAmount
   // and historical groupProfiles. Since together these change across multiple
   // locations in the code, it involves less 'code smell' to do it this way.
   // see historical/group.js for the ugly way of doing it.
   const mincomeAmount = getters.groupMincomeAmount
-  const groupProfiles = getters.groupProfiles
-  const currentIncomeDistribution = []
-  for (const username in groupProfiles) {
-    const profile = groupProfiles[username]
-    const incomeDetailsType = profile && profile.incomeDetailsType
-    if (incomeDetailsType) {
-      const adjustment = incomeDetailsType === 'incomeAmount' ? 0 : mincomeAmount
-      const amount = adjustment + profile[incomeDetailsType]
-      currentIncomeDistribution.push({
-        name: username,
-        amount: saferFloat(amount)
-      })
-    }
-  }
-  let dist = incomeDistribution(currentIncomeDistribution, mincomeAmount)
+  const distributionEvents = getters.currentGroupState.distributionEvents
```

Comment:
I don't think this is a big deal, but whenever you have direct access to `state`, just access it directly via that, instead of calling `getters.currentGroupState`.

Reply:
Actually, ignore that comment. In fact, using `state` at all in this function will cause problems. So it should probably be removed from the parameters, and only accessed the way you are accessing it.

Reply:
Ok.